File: src/main/java/com/gusrylmubarok/reddit/backend/model/Subreddit.java
package com.gusrylmubarok.reddit.backend.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import javax.validation.constraints.NotBlank;
import java.time.Instant;
import java.util.List;

@Entity
@Table(name = "subreddit")
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Subreddit {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "name")
    @NotBlank(message = "Community name is required!")
    private String name;

    @Column(name = "description")
    @NotBlank(message = "Description is required!")
    private String description;

    @Column(name = "created_date")
    private Instant createdDate;

    @ManyToOne(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    @JoinColumn(name = "user_id", referencedColumnName = "id")
    private User user;

//    @OneToMany(mappedBy = "subreddit")
//    private List<Post> posts;

}

File: src/main/kotlin/com/reddot/data/entity/Subreddit.kt
package com.reddot.data.entity

import jakarta.persistence.*
import java.util.*

@Entity
@Table(name = "subbreddit")
data class Subreddit(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private val id: Long,

    @Column(name = "name")
    private var name: String,

    @Column(name = "description")
    private var description: String,

    @OneToMany(fetch = FetchType.LAZY)
    @JoinColumn(name = "posts", referencedColumnName = "id")
    private var posts: List<Post>,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "userId", referencedColumnName = "id")
    private var user: User,

    @Column(name = "created_at")
    private var createdAt: Date,

    @Column(name = "updated_at")
    private var updatedAt: Date?
)
